storage/innobase/xtrabackup/src/xbstream.cc

```diff
@@ -516,6 +516,30 @@ static void *extract_worker_thread_func(void *arg) {
       ctxt->filehash.erase(entry->path);
       file_entry_free(entry);
       pthread_mutex_unlock(ctxt->mutex);
+      /*
+       * no need for mutex here. At this point, we are guarantee that all other
+       * threads have completed its work with this file
+       */
+      if (opt_decompress && ctxt->ds_ctxt->fs_support_punch_hole &&
+          (is_compressed_suffix(chunk.path) ||
+           is_encrypted_and_compressed_suffix(chunk.path))) {
```

Comment:
will try to do punch hole on non-InnoDB files? they will have the same extension? :thinking:

Reply:
Non InnoDB files will fail later at restore_sparseness function:

https://github.com/altmannmarcelo/percona-xtrabackup/blob/PXB-2890-8.0/storage/innobase/xtrabackup/src/file_utils.cc#L218